frontend/src/components/GameBoard/GameBoard.functions.ts

```diff
@@ -204,3 +203,26 @@ export const extendBoard = (row: number, column: number) => {
     leftColumn -= 1;
   }
 };
+
+//================== MEEPLE PLACEMENT =====================//
+
+export const placeMeeple = (location: Locations, tile: Tile) => {
+  const currentPlayer = rootStore.playersStore.getCurrentPlayer()!;
+  const availableProjects = rootStore.projectStore.getAvailableProjects(tile);
+  if (
+    rootStore.gameStore.currentPhase === GamePhases.MEEPLE_PLACEMENT &&
+    currentPlayer.getMeepleCount() > 0 &&
+    availableProjects
+  ) {
+    const meeple = currentPlayer.getMeeple()!;
+    const projectOfLocation = availableProjects.find((p: Project) => p.type === location);
+    if (projectOfLocation) {
+      projectOfLocation.meeples.push(meeple);
+    }
+  } else if (availableProjects) {
+    console.log('no meeples available');
+    //end of turn
+  } else {
+    console.log('no available projects');
+  }
```

Comment:
It is hard to read and debug. Also, we do not expect `no available projects` (no dropdown shown if it happens) 
We can dispatch invalid move event here
```js
if (!availableProjects) {
  //dispatch and return 
}

if (!meeples) {
  //dispatch and return
}


// and at this point, when all invalid scenarios are checked, we can simply do the placement action
```